### PR TITLE
Set up for integration tests for smart contracts

### LINKED ISSUE #35: Set up for integration tests for smart contracts
ℹ️ This issue is part of Epic #49 

## Objective

Set up the test environment to test contracts connection and any other code file needed to achieve coverage in a good percentage.  

## Checklist

- [x] The smart contract connection is tested from the testnet
- [ ] There are integration tests for all contracts created so far in the project.

### PR DESCRIPTION
**Pull request by LauraVilladakommit**
## Description

We add the integration test set-up to create all the integration tests missing for Leishmaniasis

Fixes #35 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: integration-tests/envVariables.json.example [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "MAIN_ACCOUNT": "ok-medicines.testnet",
```
Comment (jcamilocp) [inline integration-tests/envVariables.json.example:2]: Question: can't we use the same `contracts.env` file for this?

File: integration-tests/src/miltefosine.ava.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+const CONTRACT_ACCOUNT = process.env.MILTEFOSINE_CONTRACT;
+
+const BLOCK_ID = process.env.MILTEFOSINE_BLOCK_ID;
+
+const test = anyTest as TestFn<{
+  worker: Worker;
+  accounts: Record<string, NearAccount>;
+}>;
+
+test.beforeEach(async (t) => {
```
Comment (jcamilocp) [inline integration-tests/src/miltefosine.ava.ts:13]: I see a lot of code duplicated. Can we refactor the code to prevent that?

**Review by jcamilocp — CHANGES_REQUESTED** [review]

File: integration-tests/package.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "ava-testing",
+  "version": "1.0.0",
+  "license": "(MIT AND Apache-2.0)",
```
Comment (miguelnietoa) [inline integration-tests/package.json:4]: Why is the license "(MIT AND Apache-2.0)"?

File: integration-tests/envVariables.json.example (hunk 1 above)
Comment (LauraVilladakommit) [reply integration-tests/envVariables.json.example:2]: @jcamilocp, No, we can't. We tried to do it in several ways but ava does not recognize .env files to do the configuration set-up in `ava.config.js`. The only way to use env variables in the test environment is like this.

File: integration-tests/src/miltefosine.ava.ts (hunk 2 above)
Comment (LauraVilladakommit) [reply integration-tests/src/miltefosine.ava.ts:13]: @jcamilocp We tried also to define the contract just with a constant when importing to not use so many assignments but ava works with that t variable as an argument define in the ava context, this means it is not possible to pass the contract const like the test argument.

File: integration-tests/package.json (hunk 3 above)
Comment (LauraVilladakommit) [reply integration-tests/package.json]: I changed it to Apache-2.0

File: integration-tests/src/miltefosine.ava.ts (hunk 2 above)
Comment (jcamilocp) [reply integration-tests/src/miltefosine.ava.ts:16]: But we can define the function inside the before and after blocks in another file and then use them here, am I wrong?

File: integration-tests/src/miltefosine.ava.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  });
+
+  // Save state for test runs, it is unique for each test
+  t.context.worker = worker;
+  t.context.accounts = {root, contract };
+});
+
+test.afterEach.always(async (t) => {
+  await t.context.worker.tearDown().catch(error => {
+    console.log('Failed to tear down the worker:', error);
```
Comment (jcamilocp) [inline integration-tests/src/miltefosine.ava.ts:36]: Let's solve those warnings

File: package.json [hunk 5]
```diff
@@ -5,8 +5,8 @@
   "scripts": {
     "start": "cd okdoseweb && yarn start",
     "build": "cd okdoseweb && yarn build",
-    "postinstall": "cd okdose && yarn install && cd ../ && cd okdoseweb && yarn install",
-    "lint": "cd okdose && yarn eslint --cache . && cd .. && cd okdoseweb && yarn eslint --cache .",
+    "postinstall": "cd okdose && yarn install && cd ../ && cd okdoseweb && yarn install && cd ../ && cd integration-tests && yarn install",
+    "lint": "cd okdose && yarn eslint --cache . && cd .. && cd okdoseweb && yarn eslint --cache . && cd .. && cd integration-tests && yarn eslint --cache .",
     "lint:fix": "cd okdose && yarn eslint --fix . && cd .. && cd okdoseweb && yarn eslint --fix .",
```
Comment (jcamilocp) [inline package.json:10]: Should we modify this too?

File: .env.example [hunk 6]
```diff
@@ -0,0 +1,17 @@
+MAIN_ACCOUNT="ok-medicines.testnet"
+CONTRACT_METHOD="get_dose"
+
+NMETIL_CONTRACT="prescription.ok-medicines.testnet"
```
Comment (jcamilocp) [inline .env.example:4]: I think we should keep the .env.example with fake values, even if we're providing these values to all contributors

File: package.json (hunk 5 above)
Comment (LauraVilladakommit) [reply package.json:10]: If we want to run the integration tests in the CI, we will need to define the scripts here in the project root.

File: .env.example (hunk 6 above)
Comment (LauraVilladakommit) [reply .env.example:4]: @jcamilocp So are we going to write this info in the README to ask the contributors to write it in the .env file?

Comment (jcamilocp) [reply .env.example:4]: Yes, I think it'll be the best

File: package.json (hunk 5 above)
Comment (jcamilocp) [reply package.json:10]: We can review how to run the integration test from the CI. I think it's very important to add it to the flow

File: integration-tests/package.json [hunk 7]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "ava-testing",
+  "version": "1.0.0",
```
Comment (jcamilocp) [inline integration-tests/package.json:3]: Is this version correct?

Comment (jcamilocp) [reply integration-tests/package.json:3]: I think we're using 0.1.0 for this release.

**Review by jcamilocp — APPROVED** [review]
LGTM! Let's plan the CI for a new task

**Pull request merged**